src/Persistence/Proxy/CreatedObjectsTracker.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Zenstruck\Foundry\Persistence\Proxy;
+
+use WeakReference;
+
+use function Zenstruck\Foundry\Persistence\refresh;
+
+final class CreatedObjectsTracker
+{
+    /** @var list<\WeakReference<object>> */
+    private static $buffer = [];
+
+    public static function add(object $object): void
+    {
+        self::$buffer[] = \WeakReference::create($object);
+    }
+
+    public static function proxifyObjects(): void
+    {
+        self::$buffer = array_values(
+            array_map(
+                static function (WeakReference $weakRef) {
+                    $object = $weakRef->get() ?? throw new \LogicException('Object cannot be null.');
+                    $clone = clone $object;
+                    (new \ReflectionClass($object))->resetAsLazyProxy($object, fn() => refresh($clone));
```

Comment:
cloning the object is even easier in order to be able to refresh it!

as far as I can tell, it does not create any "identity problem"